app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args) -> Callable:
+        if args in storage:
```

Comment:
Using mutable types like lists or dictionaries as keys in a dictionary will raise a `TypeError` because they are not hashable. Ensure that `args` contains only immutable types, such as tuples of integers or strings, to avoid this issue.